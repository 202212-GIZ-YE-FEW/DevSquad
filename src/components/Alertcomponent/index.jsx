import React from "react";

import "animate.css";

const Alertcomponent = ({ type, message, icon }) => {
    const getAlertColor = () => {
        switch (type) {
            case "success":
                return "bg-green-500 text-white";
            case "warning":
                return "bg-yellow-500 text-white";
            case "error":
                return "bg-red-500 text-white";
            default:
                return "bg-gray-500 text-white";
        }
    };

    return (
        <div
            className={`fixed top-0 right-0 z-10 mx-2 p-4 max-w-4xl flex justify-center items-center space-x-2 shadow rounded animate__animated animate__backInRight ${getAlertColor()}`}
        >
            {/* :::alert message */}
            <p className='font-medium font-Rubik text-lg'>{message}</p>
            <p>{icon}</p>
        </div>
    );
};

export default Alertcomponent;
