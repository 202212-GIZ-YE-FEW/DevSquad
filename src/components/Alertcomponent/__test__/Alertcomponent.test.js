import renderer from "react-test-renderer";

import Alertcomponent from "../index";

it("renders correctly", () => {
    const tree = renderer.create(<Alertcomponent />).toJSON();
    expect(tree).toMatchSnapshot();
});
