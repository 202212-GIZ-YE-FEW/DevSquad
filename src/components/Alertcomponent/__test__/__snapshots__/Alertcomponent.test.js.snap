// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fixed top-0 right-0 z-10 mx-2 p-4 max-w-4xl flex justify-center items-center space-x-2 shadow rounded animate__animated animate__backInRight bg-gray-500 text-white"
>
   
  <p
    className="font-medium font-Rubik text-lg"
  />
</div>
`;
